feat(windows): add SDIO driver step

niStee opened the pull request:
## What does this PR do

- add a Windows-only Snappy Driver Installer Origin step that auto-detects SDIO, generates scripts for dry-run/interactive/automatic flows, and keeps the console output concise via `print_info`
- gate the step behind new `enable_sdio`/`sdio_path` config knobs, document them in `config.example.toml`, and ensure the example config keeps risky defaults disabled
- localize new user-facing text, restructure the Windows step module (`windows/mod.rs`), and wire `Step::Sdio` into the runner/default Windows step list (including a clearer "Update WSL" label)

## Standards checklist

- [x] The PR title is descriptive
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself
- [x] If this PR introduces new user-facing messages they are translated

## For new steps

- [x] *Optional:* Topgrade skips this step where needed
- [x] *Optional:* The `--dry-run` option works with this step
- [x] *Optional:* The `--yes` option works with this step if it is supported by the underlying command

## Testing

- cargo fmt
- cargo clippy
- cargo test
- /home/nils/.venvs/pre-commit/bin/pre-commit run --all-files (from WSL, skipping gitleaks per Windows limitation)
- cargo run -- --dry-run --only sdio --config tmp/sdio_topgrade_config.toml
- cargo run -- --only sdio --config tmp/sdio_topgrade_config.toml
- cargo run -- --yes --only sdio --config tmp/sdio_topgrade_config.toml
- cargo run -- --only sdio --config tmp/sdio_topgrade_config.toml -v
- cargo run -- --yes --only sdio --config tmp/sdio_topgrade_config.toml -v

niStee committed:
feat(windows): add SDIO driver step

niStee commented:
@GideonBear, please take a look.

niStee committed:
chore(windows): polish SDIO UX

niStee committed:
chore(windows): silence clippy warning

niStee committed:
Merge main into split/windows-sdio-core and update Windows steps

niStee committed:
Merge branch 'main' into split/windows-sdio-core

GideonBear commented:
Hi, sorry for the wait.
Could you send me an example of what it looks like for the step to run? E.g. can you install an outdated driver, run `topgrade --only sdio`, and show that it updates, and updated the driver, with screenshots or a screencap?
I'm not going to test SDIO myself, but I want to know what the UX would be like.

niStee committed:
Merge remote-tracking branch 'upstream/main' into pr/niStee/1338

# Conflicts:
#	locales/app.yml
#	src/config.rs

niStee committed:
fix(pre-commit): remove unnecessary blank line in configuration

niStee committed:
fix(sdio): update script commands to improve driver selection and installation process

Improves SDIO driver selection and installation control

Updates SDIO script commands to enhance the driver update process.

Adds `newer` to the `select` command, ensuring that the tool prioritizes newer drivers in addition to missing or better ones.

Explicitly controls installation enabling via `enableinstall off` during analysis and `enableinstall on` during actual installation. This prevents unintended installations during dry-runs or analysis and ensures proper execution during the installation phase.

niStee committed:
feat(windows): expand SDIO configuration support
